File: client/extract.py
import json

log_path = r'C:\Users\Vikas\.gemini\antigravity-ide\brain\6ec93dcb-4c99-4b7b-9d0d-ec77af087e00\.system_generated\logs\transcript_full.jsonl'
with open(log_path, 'r', encoding='utf-8') as f:
    for line in f:
        try:
            data = json.loads(line)
            content = data.get('content', '')
            if content and '<svg width="1440" height="2315"' in content:
                start = content.find('<svg width="1440"')
                end = content.find('</svg>', start)
                if start >= 0 and end > start:
                    svg_content = content[start:end+6]
                    with open(r'C:\react project\figmversion3\client\src\assets\journey_steps.svg', 'w', encoding='utf-8') as out:
                        out.write(svg_content)
                    print(f'Success! Extracted {len(svg_content)} characters.')
                    break
        except Exception as e:
            pass

File: client/extract.js
const fs = require('fs');
const readline = require('readline');

const logPath = 'C:\\Users\\Vikas\\.gemini\\antigravity-ide\\brain\\6ec93dcb-4c99-4b7b-9d0d-ec77af087e00\\.system_generated\\logs\\transcript_full.jsonl';
const outPath = 'C:\\react project\\figmversion3\\client\\src\\assets\\journey_steps.svg';

async function processLineByLine() {
  const fileStream = fs.createReadStream(logPath);
  const rl = readline.createInterface({
    input: fileStream,
    crlfDelay: Infinity
  });

  for await (const line of rl) {
    try {
      const data = JSON.parse(line);
      const content = data.content;
      if (content && typeof content === 'string' && content.includes('<svg width="1440" height="2315"')) {
        const start = content.indexOf('<svg width="1440"');
        const end = content.indexOf('</svg>', start);
        if (start >= 0 && end > start) {
          const svgContent = content.substring(start, end + 6);
          fs.writeFileSync(outPath, svgContent);
          console.log(Success! Extracted  characters.);
          return;
        }
      }
    } catch (e) {
      // ignore JSON parse error
    }
  }
}

processLineByLine();
